go-apps/meep-rnis/server/rnis.go

```diff
@@ -568,7 +566,7 @@ func checkReNotificationRegisteredSubscriptions(appId string, assocId *Associate
 				match = false
 			}
 
-			if match && (((sub.FilterCriteria.Plmn == nil) || (sub.FilterCriteria.Plmn != nil && (newPlmn != nil && newPlmn.Mnc == sub.FilterCriteria.Plmn.Mnc && newPlmn.Mcc == sub.FilterCriteria.Plmn.Mcc))) && (oldPlmn == nil || oldCellId == "")) {
+			if match && (((sub.FilterCriteria.Plmn == nil) || (sub.FilterCriteria.Plmn != nil && (newPlmn != nil && newPlmn.Mnc == sub.FilterCriteria.Plmn.Mnc && newPlmn.Mcc == sub.FilterCriteria.Plmn.Mcc))) && (oldPlmn == nil || oldCellId == "" || erabId != -1)) {
```

Comment:
It is difficult to understand what is going-on in the if statement above... it would be a good idea to replace the whole block with this `match = myNewFunction(param1, param2, param3, etc.)` and expand the logic (adding a few comments)

Reply:
i agree that this could be made simpler through using a function, but this code will need rework when supporting V2 because subscription changed and it has effects on the filters, so i suggest reworking it when implementing v2.

Reply:
I insist, please clarify this section of code as we may not support V2 for some time.